src/waveform/renderers/waveformwidgetrenderer.cpp

```diff
@@ -288,3 +288,31 @@ void WaveformWidgetRenderer::setTrack(TrackPointer track) {
         m_rendererStack[i]->onSetTrack();
     }
 }
+
+std::optional<CuePointer> WaveformWidgetRenderer::getCueAtPoint(QPoint point) {
+    WaveformMarkPointer pSelectedMark;
+    for (auto pMark : m_markBoundaries.keys()) {
+        const auto labelRectangle = m_markBoundaries[pMark];
+        if (labelRectangle.contains(point)) {
+            pSelectedMark = pMark;
+            break;
+        }
+    }
+    if (!pSelectedMark) {
+        return std::nullopt;
+    }
+
+    CuePointer pSelectedCue;
+    QList<CuePointer> cueList = getTrackInfo()->getCuePoints();
+    for (const auto& pCue : cueList) {
+        if (pCue->getHotCue() == pSelectedMark->getHotCue()) {
```

Comment:
Do we need to skip *cue points* with type != mixxx::CueType::HotCue? pCue->getHotCue() is -1 for many other types.

Maybe like so:
```
VERIFY_OR_DEBUG_ASSERT(pCue->getType() == mixxx::CueType::HotCue) {
    continue;
}
```

Reply:
Due to the Serato/Rekordbox import loop cues can also have a hotcue number now.

Reply:
Then we need to compare both type and number here? Are there any rules regarding uniqueness of this number in relation to the type?

I haven't checked yet how the remaining code deals with this situation.

Reply:
The numbers need to be unique or strange things might happen. Currently the only way to get loop cues with numbers is via import from Rekordbox/Serato, so there shouldn't be an issue.

Reply:
> Do we need to skip _cue points_ with type != mixxx::CueType::HotCue? pCue->getHotCue() is -1 for many other types.

Currently, only hotcues seem to be having a context menu.